s4cmb/tod.py

```diff
@@ -616,31 +690,55 @@ def compute_simpolangle(self, ch, parallactic_angle, polangle_err=False):
         ...     parallactic_angle=np.array([np.pi] * tod.nsamples))
         >>> assert angles2 is not None
         """
-        if not polangle_err:
-            ang_pix = (90.0 - self.intrinsic_polangle[ch]) * d2r
+        
+        if polangle_err:
+            state_for_polang = np.random.RandomState(self.seed_pa)#+ch) # no need to add ch because already different for each channel as this is for all detectors.
+            
 
-            ## Demodulation or pair diff use different convention
-            ## for the definition of the angle.
-            if not hasattr(self, 'dm'):
-                pol_ang = parallactic_angle + ang_pix + 2.0 * self.hwpangle
+            if self.coherent_pa:
+                if ch==0:
+                    print('All angle shifts are {}.'.format(self.pa_mu))
+                dpolang = np.ones_like(dpolang)*self.pa_mu # in deg
```

Comment:
`np.ones_like(dpolang) `
should be replace with
`np.ones(len(self.intrinsic_polangle))`

Reply:
Can you push the change?

Reply:
Should be fixed in the pull request now, hopefully.